mm2src/adex_cli/src/scenarios/mm2_proc_mng.rs

```diff
@@ -0,0 +1,174 @@
+    let s = System::new_all();
+
+    s.processes()
+        .iter()
+        .filter(|(_, process)| process.name() == pname)
+        .map(|(pid, _)| pid.as_u32())
+        .collect()
+}
+
+fn get_mm2_binary_path() -> Result<PathBuf, ()> {
+    let mut dir = env::current_exe().map_err(|error| {
+        error!("Failed to get current binary dir: {error}");
+    })?;
+    dir.pop();
+    dir.push(MM2_BINARY);
+    Ok(dir)
+}
+
+pub fn start_process(mm2_cfg_file: &Option<String>, coins_file: &Option<String>, log_file: &Option<String>) {
+    let mm2_binary = match get_mm2_binary_path() {
+        Err(_) => return,
+        Ok(path) => path,
+    };
+
+    if let Some(mm2_cfg_file) = mm2_cfg_file {
+        info!("Set env MM_CONF_PATH as: {mm2_cfg_file}");
+        env::set_var("MM_CONF_PATH", mm2_cfg_file);
+    }
+    if let Some(coins_file) = coins_file {
+        info!("Set env MM_COINS_PATH as: {coins_file}");
+        env::set_var("MM_COINS_PATH", coins_file);
+    }
+    if let Some(log_file) = log_file {
+        info!("Set env MM_LOG as: {log_file}");
+        env::set_var("MM_LOG", log_file);
+    }
+    start_process_impl(mm2_binary);
+}
+
+#[cfg(unix)]
+pub fn start_process_impl(mm2_binary: PathBuf) {
+    let mut command = Command::new(&mm2_binary);
+    let program = mm2_binary
+        .file_name()
+        .map_or("Undefined", |name: &OsStr| name.to_str().unwrap_or("Undefined"));
+
+    match daemon(true, true) {
+        Ok(Fork::Child) => {
+            command.output().expect("failed to execute process");
+        },
+        Ok(Fork::Parent(pid)) => {
+            info!("Successfully started: {program:?}, forked pid: {pid}");
+        },
+        Err(error) => error!("Failed to fork a process: {error}"),
+    }
+}
+
+#[cfg(windows)]
+pub fn start_process_impl(mm2_binary: PathBuf) {
+    let program = mm2_binary.to_str();
+    if program.is_none() {
+        error!("Failed to cast mm2_binary to &str");
+        return;
+    }
+    let program = CString::new(program.unwrap());
+    if let Err(error) = program {
+        error!("Failed to construct CString program path: {error}");
+        return;
+    }
+
+    let mut startup_info: STARTUPINFOA = unsafe { mem::zeroed() };
+    startup_info.cb = size_of::<STARTUPINFOA>() as u32;
+    let mut process_info: PROCESS_INFORMATION = unsafe { mem::zeroed() };
+
+    let result = unsafe {
+        CreateProcessA(
+            null(),
+            program.unwrap().into_raw() as *mut i8,
+            std::ptr::null_mut(),
+            std::ptr::null_mut(),
+            0,
+            0,
+            std::ptr::null_mut(),
+            std::ptr::null(),
+            &mut startup_info as &mut STARTUPINFOA,
+            &mut process_info as *mut PROCESS_INFORMATION,
+        )
+    };
+
+    match result {
+        0 => error!("Failed to start: {MM2_BINARY}"),
+        _ => {
+            info!("Successfully started: {MM2_BINARY}");
+        },
+    }
```

Comment:
I personally prefer if conditions in such cases
```suggestion
    if result == 0 {
        error!("Failed to start: {MM2_BINARY}")
    } else {
        info!("Successfully started: {MM2_BINARY}")
    }
```

Reply:
How about this:

```rust
    match result {
        0 => error!("Failed to start: {MM2_BINARY}"),
        _ => info!("Successfully started: {MM2_BINARY}"),
    }
```

Reply:
Yeah that looks better.

I usually prefer `if` if I only care about 1 or 2 results of the condition, and pattern matching for the rest. This is not a standard or something. Just my honest opinion.

Reply:
Okay I understand what you mean, I thought the problem was indentation. Yes if we are just checking for one case I also think that if else statements are more intuitive and easier to understand.

Reply:
Thank you, guys! done.